File: frontend/src/types/job.ts
export interface JobPostSummary {
  id: string;
  company_name: string | null;
  company_type: string | null;
  status: string;
  title: string;
  job_category: string | null;
  employment_type: string | null;
  shift_type: string | null;
  salary_type: string | null;
  salary_min: number | null;
  salary_max: number | null;
  location_code: string | null;
  location_detail: string | null;
  close_at: string | null;
  published_at: string | null;
  view_count: number;
}

export interface JobPostDetail {
  id: string;
  company_id: string;
  company_name: string | null;
  company_type: string | null;
  status: string;
  title: string;
  body: string | null;
  job_category: string | null;
  department: string | null;
  specialty: string | null;
  employment_type: string | null;
  shift_type: string | null;
  salary_type: string | null;
  salary_min: number | null;
  salary_max: number | null;
  location_code: string | null;
  location_detail: string | null;
  contact_name: string | null;
  contact_visible: boolean;
  close_at: string | null;
  published_at: string | null;
  view_count: number;
  created_at: string;
  updated_at: string;
}

export interface JobListResponse {
  items: JobPostSummary[];
  page: number;
  size: number;
  total: number;
}

export interface JobSearchParams {
  keyword?: string;
  location_code?: string;
  job_category?: string;
  shift_type?: string;
  employment_type?: string;
  salary_min?: number;
  sort?: string;
  page?: number;
  size?: number;
}

File: backend/app/schemas/job.py
from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel, Field


class JobPostCreate(BaseModel):
    title: str = Field(..., min_length=1, max_length=300)
    body: str | None = None
    job_category: str | None = None
    department: str | None = None
    specialty: str | None = None
    employment_type: str | None = None
    shift_type: str | None = None
    salary_type: str | None = None
    salary_min: int | None = None
    salary_max: int | None = None
    location_code: str | None = None
    location_detail: str | None = None
    contact_name: str | None = None
    contact_visible: bool = False
    close_at: date | None = None


class JobPostUpdate(BaseModel):
    title: str | None = Field(None, min_length=1, max_length=300)
    body: str | None = None
    job_category: str | None = None
    department: str | None = None
    specialty: str | None = None
    employment_type: str | None = None
    shift_type: str | None = None
    salary_type: str | None = None
    salary_min: int | None = None
    salary_max: int | None = None
    location_code: str | None = None
    location_detail: str | None = None
    contact_name: str | None = None
    contact_visible: bool | None = None
    close_at: date | None = None


class JobPostRead(BaseModel):
    id: str
    company_id: str
    company_name: str | None = None
    company_type: str | None = None
    status: str
    title: str
    body: str | None = None
    job_category: str | None = None
    department: str | None = None
    specialty: str | None = None
    employment_type: str | None = None
    shift_type: str | None = None
    salary_type: str | None = None
    salary_min: int | None = None
    salary_max: int | None = None
    location_code: str | None = None
    location_detail: str | None = None
    contact_name: str | None = None
    contact_visible: bool = False
    close_at: date | None = None
    published_at: datetime | None = None
    view_count: int = 0
    created_at: datetime
    updated_at: datetime


class JobPostSummary(BaseModel):
    id: str
    company_name: str | None = None
    company_type: str | None = None
    status: str
    title: str
    job_category: str | None = None
    employment_type: str | None = None
    shift_type: str | None = None
    salary_type: str | None = None
    salary_min: int | None = None
    salary_max: int | None = None
    location_code: str | None = None
    location_detail: str | None = None
    close_at: date | None = None
    published_at: datetime | None = None
    view_count: int = 0


class JobListResponse(BaseModel):
    items: list[JobPostSummary]
    page: int
    size: int
    total: int


class JobSitemapEntry(BaseModel):
    id: str
    updated_at: datetime
